### ISSUE TITLE
Cannot connect outbound email to Microsoft Oauth

### ISSUE THREAD

**Issue by congnvp** [salesagility/SuiteCRM#10086]
#### Issue

As I know, SMTP is not anymore supported in Microsoft 365. But there's still no option to authenticate outbound and notification email using Oauth in Microsoft 365.

#### Steps to Reproduce

1.
2.
3.
4.

#### Your Environment

* SuiteCRM Version used:
* Browser name and version (e.g. Chrome Version 51.0.2704.63 (64-bit)):
* Environment name and version (e.g. MySQL, PHP 7):
* Operating System and version (e.g Ubuntu 16.04):

**Comment by chris001** [comment]
> As I know, SMTP is not anymore supported in Microsoft 365. 
> But there's still no option to authenticate outbound and notification email using Oauth in Microsoft 365.

@congnvp 1. Could you Share link to where it says SMTP is not anymore supported in Microsoft 365?
2. What makes you say Still no option to authenticate outbound and notification email using Oauth in Microsoft 365`?

**Comment by congnvp** [comment]
> > As I know, SMTP is not anymore supported in Microsoft 365. 
> > But there's still no option to authenticate outbound and notification email using Oauth in Microsoft 365.
> 
> @congnvp 1. Could you Share link to where it says SMTP is not anymore supported in Microsoft 365?
> 2. What makes you say Still no option to authenticate outbound and notification email using Oauth in Microsoft 365`?

I have read this article that basic authentication have deprecated, in the meantime, app password have deprecated too. Then I found out there's no app password option any more in "account info" page. 
[https://learn.microsoft.com/en-us/exchange/clients-and-mobile-in-exchange-online/deprecation-of-basic-authentication-exchange-online](https://learn.microsoft.com/en-us/exchange/clients-and-mobile-in-exchange-online/deprecation-of-basic-authentication-exchange-online)

Legacy-smtp still be keeped, however I cannot make it work although I followed the instructions in SharePoint PowerShell. Moreover, it is definitely not a choice in the future. 

I'm looking for some solutions like email proxies but haven't found one that works.

**Comment by chris001** [comment]
@congnvp 
> I'm looking for some solutions [...] but haven't found one that works.

The [way to fix the SuiteCRM PHPmailer outbound email, is here](https://github.com/PHPMailer/PHPMailer/wiki/Microsoft-OAuth2-SMTP-issues).

**Comment by congnvp** [comment]
> 

Nah, I have two domain run with exchange. I've working hard to findout the problem. Following the @chris001 document. But it will be very hard to user adding their email.

**Comment by congnvp** [comment]
> > As I know, SMTP is not anymore supported in Microsoft 365.
> > But there's still no option to authenticate outbound and notification email using Oauth in Microsoft 365.
> 
> @congnvp 1. Could you Share link to where it says SMTP is not anymore supported in Microsoft 365? 2. What makes you say Still no option to authenticate outbound and notification email using Oauth in Microsoft 365`?

I foundout that only Suitecrm cannot connect to microsoft365 SMTP. I tested with other's (mautic and openmediavault) it worked fine, I wonder if it is only suitecrm error? How can I fix?

**Comment by congnvp** [comment]
> @congnvp 
> Are you admin?  
> Can you go into your micrsoft 365 tenant settings.
> Settings > Org Settings > Modern Authentication. Set it to Off or Disabled or No.
> 
> ```
> In July 2020 Microsoft announced that it would disable SMTP, IMAP and other protocols 
> using Basic Authentication for all tenants that weren't using them and introduced a simpler 
> tenant-wide option for Admin to manage this 
> (Settings > Org Settings > Modern Authentication). 
> ```

I found-out the way to fix it. Phpmailer seem does not support TLS 1.3, so yesterday I turned on legacy-SMTP from office 365 then it worked fine. But hopefully suitecrm support oauth2 soon for outgoing mail, it will be more suitable for many low-tech user.

**Comment by chris001** [comment]
It looks like PHPMailer supports and uses `TLS 1.3`, as long as your `PHP` version is `7.2`+, and `openssl` version `1.1.1`+, and will use it, if Microsoft 365 server enables`TLS 1.3` also.
If you run `phpinfo();` and search for the versions of `php` and `openssl` your CRM server is using.

**Comment by chris001** [comment]
@congnvp 
What's the output of:
`php -r "phpinfo();" | grep OpenSSL`

**Comment by congnvp** [comment]
> @congnvp 
> What's the output of:
> `php -r "phpinfo();" | grep OpenSSL`

I've got this:
SSL Version => OpenSSL/3.0.2
OpenSSL support => enabled
OpenSSL Library Version => OpenSSL 3.0.2 15 Mar 2022
OpenSSL Header Version => OpenSSL 3.0.2 15 Mar 2022
Native OpenSSL support => enabled
OpenSSL => Stig Venaas, Wez Furlong, Sascha Kettler, Scott MacVicar

**Comment by chris001** [comment]
Thanks, your system has recent enough versions of PHP, PHPmailer, and `openssl` it should support `TLS 1.3`.  Anyway, it's good you got the `SMTP` outgoing mail working by enabling legacy SMTP in Microsoft 365.

More on [how to update the app to connect to Microsoft with OAuth2 tokens to send mail, instead of the deprecated way of using a password](https://github.com/PHPMailer/PHPMailer/discussions/2747).

**Comment by gassalta** [comment]
Excuse my bad translation, but did I try following these steps?
I have managed to send emails using outlook 365 by smtp

1. https://docs.suitecrm.com/admin/administration-panel/emails/microsoft-oauth-provider-howto/
2. https://docs.suitecrm.com/admin/administration-panel/emails/inboundemail-oauth-howto/
